lib/spack/spack/build_systems/python.py

```diff
@@ -135,14 +136,20 @@ def import_modules(self):
         # Some Python libraries are packages: collections of modules
         # distributed in directories containing __init__.py files
         for path in find(root, '__init__.py', recursive=True):
-            modules.append(path.replace(root + os.sep, '', 1).replace(
-                os.sep + '__init__.py', '').replace('/', '.'))
+            mod = path.replace(root + os.sep, '', 1).replace(
+                os.sep + '__init__.py', '').replace('/', '.')
+            if not re.match('[a-zA-Z0-9._]+$', mod):
+                continue
+            modules.append(mod)
 
         # Some Python libraries are modules: individual *.py files
         # found in the site-packages directory
         for path in find(root, '*.py', recursive=False):
-            modules.append(path.replace(root + os.sep, '', 1).replace(
-                '.py', '').replace('/', '.'))
+            mod = path.replace(root + os.sep, '', 1).replace(
+                '.py', '').replace('/', '.')
+            if not re.match('[a-zA-Z0-9._]+$', mod):
+                continue
+            modules.append(mod)
```

Comment:
There would be less code duplication if you append all modules to the list and then filter afterwards. I would personally use Python's `filter()` function with a lambda.

Reply:
Still waiting on this change

Reply:
Sorry, forgot to push.